diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -606,10 +618,11 @@ namespace diff_drive_controller
       }
     }
 
+    // Compute the period estimated from time:
+    const ros::Duration period_estimated = time - time_previous_;
+
     // Compute/Set control period and frequency (desired/expected or real):
-    // @todo also allow to compute the period as time - time_previous_, because
-    // they can be different
-    const double control_period    = control_period_desired_    > 0.0 ? control_period_desired_    : period.toSec();
+    const double control_period    = control_period_desired_    > 0.0 ? control_period_desired_    : (period_from_time_ ? period_estimated.toSec() : period.toSec());
     const double control_frequency = control_frequency_desired_ > 0.0 ? control_frequency_desired_ : 1.0 / control_period;
```

Comment:
I can't use ROS_ log messages or the diagnostic updater because they're NOT RT-safe.

For now I can put a check and return from the update method... or at least skip the code that assumes `period > 0`.

My intention is to make a RT diagnostic updater, which can probably be done by simply passing a RealTimePublisher.